boofuzz/raw_l3_socket_connection.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import
+
+import errno
+import socket
+import sys
+
+from future.utils import raise_
+
+from . import base_socket_connection, exception
+
+ETH_P_IP = 0x0800  # Ethernet protocol: Internet Protocol packet, see Linux <net/if_ether.h> docs for more details.
+
+
+class RawL3SocketConnection(base_socket_connection.BaseSocketConnection):
+    """BaseSocketConnection implementation for use with Raw Layer 2 Sockets.
+
+    Args:
+        interface (str): Interface to send and receive on.
+        send_timeout (float): Seconds to wait for send before timing out. Default 5.0.
+        recv_timeout (float): Seconds to wait for recv before timing out. Default 5.0.
+        ethernet_proto (int): Ethernet protocol to bind to. Defaults to ETH_P_IP (0x0800).
+        l2_dst (str): Layer2 destination address (e.g. MAC address). Default '\xFF\xFF\xFF\xFF\xFF\xFF' (broadcast)
+        packet_size (int): Maximum packet size (in bytes). Default 1500 if the underlying interface uses
+            standard ethernet for layer 2. Otherwise, a different packet size may apply (e.g. Jumboframes,
+            802.5 Token Ring, 802.11 wifi, ...) that must be specified.
+    """
+
+    def __init__(
+        self,
+        interface,
+        send_timeout=5.0,
+        recv_timeout=5.0,
+        ethernet_proto=ETH_P_IP,
+        l2_dst=b"\xff" * 6,
+        packet_size=1500,
+    ):
+        super(RawL3SocketConnection, self).__init__(send_timeout, recv_timeout)
+
+        self.interface = interface
+        self.ethernet_proto = ethernet_proto
+        self.l2_dst = l2_dst
+        self.packet_size = packet_size
+
+    def open(self):
+        self._sock = socket.socket(socket.AF_PACKET, socket.SOCK_DGRAM)
```

Comment:
Why don't we open the socket with ethernet proto like we do on raw l2?
https://github.com/jtpereyda/boofuzz/blob/c303290dcd79da2af0ffc24d56ca8203f03bad0d/boofuzz/raw_l2_socket_connection.py#L48
Might be redundant because we bind the socket later.
I ran some tests and everything works as intended. Should we still add it for consistency?

Reply:
Hm, this code came verbatim from what was done in the old SocketConnection. But, as you mentioned, I suppose we should unify this.